dq: sync ca: fix watermarks enabling

yumkam opened the pull request:
### Changelog entry <!-- a user-readable short description of the changes that goes to CHANGELOG.md and Release Notes -->

...

### Changelog category <!-- remove all except one -->

* Not for changelog (changelog entry is not required)

### Description for reviewers <!-- (optional) description for those who read this PR -->

Only worked previously in kqp, was silently ignored with dq (and inside ut)

yumkam committed:
dq: sync ca: fix WatermarkTracker passing

yumkam committed:
kqp/ca: remove unused parameter

yumkam committed:
disable now-exposed failing test

APozdniakov commented on ydb/library/yql/dq/actors/compute/dq_sync_compute_actor_base.h:
```diff
@@ -274,7 +274,7 @@ class TDqSyncComputeActorBase: public TDqComputeActorBase<TDerived, TComputeActo
         }
 
-        TaskRunner->Prepare(this->Task, limits, execCtx);
+        TaskRunner->Prepare(this->Task, limits, execCtx, &this->WatermarksTracker);
```
Выглядит, будто теперь вотермарки включаются вне зависимости от фича флага

yumkam replied:
Без фиче-флага ватермарки не расползаются из сорцов:
```
+    TasksGraph.BuildCheckpointingAndWatermarksMode(enableCheckpoints, EnableWatermarks);
```

yumkam replied:
Я ещё думал *а нельзя ли привязать передачу `&WatermarkTracker` к наличию входов с ватермарками*. К сожалению, там порядок вызова противоположный: вначале вызывается Prepare(), а потом (в `FillIoMaps`) обходим протобуф и вытягиваем из него настройки ватермарок в каналах/входах. Можно делать два прохода по протобуфам, но это как-то неуклюже

yumkam commented on ydb/library/yql/dq/actors/compute/dq_compute_actor_watermarks.h:
```diff
@@ -21,8 +21,8 @@ class TDqComputeActorWatermarks
 
-    void UnregisterAsyncInput(ui64 inputId);
-    void UnregisterInputChannel(ui64 inputId);
+    void UnregisterAsyncInput(ui64 inputId, bool silent = false);
```
Теоретически могут быть входы без ватермарок, а код финализации не знает -- этот вход был с ватермарками или без

yumkam commented on ydb/library/yql/dq/runtime/dq_input_producer.cpp:
```diff
@@ -192,23 +196,13 @@ class TDqInputUnionStreamValue : public TComputationValue<TDqInputUnionStreamVal
-
-    [[nodiscard]] bool TrySendWatermark() {
+    bool TrySendWatermark() {
         if (!WatermarksEnabled()) {
```
Нет -- в async ca тут false;
Такты можно чуть-чуть пополировать выкинув (переместив в инициализацию) проверку WatermarksStorage

yumkam committed:
runtime/dq_task_runner: disable watermarks tracker on tasks/inputs without watermarks

in practice it should be no-op (WatermarksTracker->Notify will return false on
non-registered inputs, watermarks are never generated, etc)
